### ISSUE TITLE
Fine-tune deploys stay 'building' with no publishedName — unreachable via the gateway

### ISSUE THREAD

**Issue by kreuzhofer** [kreuzhofer/dgx-manager#84]
## Problem

A deployment created through `POST /api/finetune/:id/deploy` is written once with `status: "building"` and **never updated**, so it never reaches `running` and never gets a `publishedName`.

Because the gateway builds its model list and routing allowlist from **published names**, the consequence is functional, not cosmetic:

**Every fine-tuned model is unreachable through the gateway.** It can only be used by addressing the node directly.

## Measured 2026-09-04

Two deployments of the same base architecture, made minutes apart on the same fleet:

| | spark-01 — `POST /api/deployments` | spark-04 — `POST /api/finetune/:id/deploy` |
|---|---|---|
| id | `cmtmvb5ud5azz2auhkj56zyd6` | `cmtmx6ac25dy52auh3yhkwjt0` |
| status | `running` | **`building`** |
| publishedName | `qwen3.8-27b` | **`null`** |
| createdAt | 11:25:42.613Z | 12:17:54.386Z |
| updatedAt | 11:34:51.377Z (**+9 min**) | 12:17:54.640Z (**+254 ms**) |
| endpoint on node | HTTP 200 | **HTTP 200** |

The spark-04 record was written at creation and never touched again — while the model was serving correctly the whole time.

Gateway consequence:

```
GET /v1/models            -> glm-5.3-flash, qwen3.8-27b        (merged model absent)

POST /v1/chat/completions {"model":"qwen3.8-sql-500step-r2"}
  -> {"error":{"message":"The model 'qwen3.8-sql-500step-r2' does not exist.",
               "type":"invalid_request_error","code":"model_not_found"}}
```

Direct to the node works fine:
```
GET http://192.168.44.39:8000/v1/models
  -> {"id":"qwen3.8-sql-500step-r2","root":"/mnt/tank/outputs/cmtmt9dbd57gi2auhok8cptou/merged"}
```

## Why it matters

Per `CONTEXT.md`, the gateway exists so that *"clients should need one address and one name, not a node IP and a port that changes when a deployment moves"*. A fine-tuned model that only answers on a node IP defeats exactly that, and it is the deployment class most likely to be consumed by downstream apps (chat3d et al.) rather than by a human with SSH access.

It also silently breaks the **published name** contract in `CONTEXT.md`: a published name is supposed to be *"established when the deployment starts serving"*. Here the deployment is serving and no name is ever established.

## Where to look

The regular deploy path clearly transitions `building → running` and discovers/persists the published name; the fine-tune deploy path does not. Likely the fine-tune handler in `packages/server/src/routes/finetune.ts` (`POST /:id/deploy`, around line 1084) never subscribes to, or never persists, the readiness/published-name update that `routes/deployments.ts` handles.

Note the related-but-distinct [#81](https://github.com/kreuzhofer/dgx-manager/issues/81): fine-tune **job** records also never update their status/phase/step/loss. Same smell — the fine-tune paths appear to skip the record-update step the regular paths perform. Worth checking whether one fix addresses both.

## Suggested test

Integration test asserting that after a fine-tune deploy reports ready, the deployment row has `status: running` **and** a non-null `publishedName`, and that `GET /v1/models` includes it. Existing coverage evidently asserts the deploy is *created*, not that it becomes reachable.

Found while working [#79](https://github.com/kreuzhofer/dgx-manager/issues/79); spotted by Daniel noticing the dashboard still said "building" for a model that was demonstrably serving.

**Comment by kreuzhofer** [comment]
## Wider impact: `port` and `clusterMode` are also never set, which strips the dashboard actions

Daniel flagged that the deployment shows none of the usual row affordances. Root-caused to the same defect: the record is written once and never updated, so **three** fields stay at their initial values, not just `status`.

| field | spark-04 (`/api/finetune/:id/deploy`) | spark-01 (`/api/deployments`) |
|---|---|---|
| status | `building` | `running` |
| **port** | **`null`** | `8000` |
| **clusterMode** | **`false`** | `true` |

### What that hides in the UI

From `packages/dashboard/app/deployments/page.tsx`:

- **`★ head` badge** — `role: d.clusterMode ? "head" : "solo"` (line 1307). `clusterMode=false` → role `"solo"` → badge suppressed (line 1429).
- **`API`, `Claude`, `Benchmark` buttons** — each gated on `d.status === "running" && d.port` (lines ~1455, ~1465, ~1477). The fine-tune deploy fails **both** conditions, so all three vanish.

So the operator loses the API link, the Claude launcher, **and the ability to start a benchmark** against a fine-tuned model from the UI. Benchmarking the thing you just trained is arguably the primary reason to deploy it, which makes this more than cosmetic on top of the gateway breakage already described above.

### Not explained by this

**`Logs`** is gated only on `!isWorker` (line 1432), and a solo deployment is not a worker, so by the code it should still render — `isFinetune` only affects edit-form copy, not the action row. If Logs is genuinely absent rather than present-but-empty, there is a second cause and it needs its own look.

### Note on `clusterMode`

Worth deciding what `clusterMode` should be for a single-node deploy. spark-01 was created with a single `nodeId` and still has `clusterMode=true` — which is why it shows `★ head` despite being solo. So the fix is not simply "set clusterMode=true on fine-tune deploys"; the two paths disagree about what the flag means, and the badge arguably should not appear for either single-node deployment.

### Fix should cover

1. `status` → `running` on readiness
2. `publishedName` discovered and persisted (gateway routing — see above)
3. `port` persisted (unblocks API/Claude/Benchmark)
4. `clusterMode` set consistently with the regular deploy path, once its intended meaning for solo deploys is settled

A test asserting a fine-tune deploy ends up **indistinguishable from a regular deploy** in every field the dashboard and gateway read would catch all four at once.

**Comment by kreuzhofer** [comment]
**Decision from Daniel:** `★ head` should **not** show on single-node deployments.

So the `clusterMode` half of this issue resolves to *two* changes, not one:

1. The fine-tune deploy path should set `clusterMode` consistently with the regular path (part of the "never updated" defect above).
2. The badge itself should be gated on an actual multi-node cluster, not on `clusterMode` alone. Today `dgx-spark-01` was deployed with a single `nodeId`, carries `clusterMode=true`, and displays `★ head` while being solo — that is the bug in the other direction.

Concretely, `role` at `packages/dashboard/app/deployments/page.tsx:1307` is:

```ts
role: d.clusterMode ? "head" : "solo"
```

which makes "head" mean "clusterMode is set" rather than "this deployment spans more than one node". Gating on the presence of worker `clusterNodes` (i.e. `d.clusterNodes?.some(cn => cn.role === "worker")`) would give the badge its intended meaning and fix both directions at once — the false badge on solo deploys and the missing one on genuine clusters.

Worth confirming against a real 2-node deploy: `cmtmtc64857kx2auheovbmrla` (glm-5.3-flash on spark-02 head + spark-03 worker) is the case that *should* keep the badge.
